jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/xloader/ProcIngestDataX.java

```diff
@@ -43,39 +43,51 @@
 import org.apache.jena.system.progress.ProgressMonitor;
 import org.apache.jena.system.progress.ProgressMonitorOutput;
 import org.apache.jena.tdb2.DatabaseMgr;
+import org.apache.jena.tdb2.params.StoreParams;
+import org.apache.jena.tdb2.params.StoreParamsConst;
 import org.apache.jena.tdb2.solver.stats.Stats;
 import org.apache.jena.tdb2.solver.stats.StatsCollectorNodeId;
 import org.apache.jena.tdb2.store.DatasetGraphTDB;
 import org.apache.jena.tdb2.store.NodeId;
 import org.apache.jena.tdb2.store.nodetable.NodeTable;
 import org.apache.jena.tdb2.store.nodetupletable.NodeTupleTable;
 import org.apache.jena.tdb2.store.value.DoubleNode62;
+import org.apache.jena.tdb2.sys.DatabaseConnection;
 import org.apache.jena.tdb2.sys.TDBInternal;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Create the Node table and write the triples/quads temporary files */
+/**
+ * Load the triples and quads to temporary files.
+ * <p>
+ * If the node table has been created ({@link ProcBuildNodeTableX}),
+ * <p>
+ * If the node table was created, this step creates the node table.
```

Comment:
The node table is something on my list-of-things-to-study (main reason why I got a job Mon-Thu; so Fridays are for OSS & having fun with interesting code). Already understand the query parsing, some of the algebra part & streaming from the JSON PR from some years ago... and the nodes/datatypes... but going to try to dig deeper into how the node table works (and maybe later transactions).

So hopefully will be able to review more thoroughly changes like this one - for now I just check if the code "makes sense" or if there's anything that looks awkward.

Reply:
Comment fixed.

The bulkloaders probably aren't the place to learn about the architecture. They trample straight though "architecture"; they are "anything goes - just get it done".  :woozy_face:

Reply:
It's nevertheless fun to get a glimpse of what's underneath of Fuseki & SPARQL in Jena :nerd_face: